cmd/harmony/main.go

```diff
@@ -309,29 +318,51 @@ func readMultiBLSKeys(consensusMultiBLSPriKey *multibls.PrivateKey, consensusMul
 		)
 		os.Exit(100)
 	}
-	if len(blsKeyFiles) > *maxBLSKeysPerNode {
-		fmt.Fprintf(os.Stderr,
-			"[Multi-BLS] maximum number of bls keys per node is %d, found: %d\n",
-			*maxBLSKeysPerNode,
-			len(blsKeyFiles),
-		)
-		os.Exit(100)
-	}
-	for _, blsKeyFile := range blsKeyFiles {
-		fullName := blsKeyFile.Name()
-		ext := filepath.Ext(fullName)
-		name := fullName[:len(fullName)-len(ext)]
-		if val, ok := keyPasses[name]; ok {
-			blsPassphrase = val
+
+	if len(awsEncryptedBLSKeyFiles) > 0 {
+		if len(awsEncryptedBLSKeyFiles) > *maxBLSKeysPerNode {
+			fmt.Fprintf(os.Stderr,
+				"[Multi-BLS] maximum number of bls keys per node is %d, found: %d\n",
+				*maxBLSKeysPerNode,
+				len(awsEncryptedBLSKeyFiles),
+			)
+			os.Exit(100)
 		}
-		blsKeyFilePath := path.Join(*blsFolder, blsKeyFile.Name())
-		consensusPriKey, err := blsgen.LoadBLSKeyWithPassPhrase(blsKeyFilePath, blsPassphrase)
-		if err != nil {
-			return err
+		for _, blsKeyFile := range awsEncryptedBLSKeyFiles {
+			blsKeyFilePath := path.Join(*blsFolder, blsKeyFile.Name())
+			consensusPriKey, err := blsgen.LoadAwsCMKEncryptedBLSKey(blsKeyFilePath, awsSettingString)
+			if err != nil {
+				return err
```

Comment:
return errors.Wrapf(err, "count not load key from path %s using settings %s", blsKeyFilePath, awsSettingString)